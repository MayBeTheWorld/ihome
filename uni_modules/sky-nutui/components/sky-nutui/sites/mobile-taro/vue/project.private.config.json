{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "feedback/pages/notify/index",
          "pathName": "feedback/pages/notify/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "business/pages/signature/index",
          "pathName": "business/pages/signature/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/collapse/index",
          "pathName": "exhibition/pages/collapse/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "NutUI-Taro"
}