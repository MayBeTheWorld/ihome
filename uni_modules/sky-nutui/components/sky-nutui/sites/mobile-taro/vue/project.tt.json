{
  "miniprogramRoot": "./",
  "projectname": "@nutui/nutui-taro-mobile",
  "appid": "touristappid",
  "setting": {
    "es6": false,
    "minified": false
  }
}
