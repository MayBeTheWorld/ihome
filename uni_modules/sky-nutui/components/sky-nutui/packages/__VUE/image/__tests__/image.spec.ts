import { config, mount } from '@vue/test-utils';
import { h, nextTick } from 'vue';
import Image from '../index.vue';
import NutIcon from '../../icon/index.vue';

beforeAll(() => {
  config.global.components = {
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

test('image render', async () => {
  const wrapper = mount(Image, {
    props: {
      src: '//img10.360buyimg.com/ling/jfs/t1/181258/24/10385/53029/60d04978Ef21f2d42/92baeb21f907cd24.jpg',
      width: '100',
      height: '100',
      showLoading: false
    }
  });
  await wrapper.find('img').trigger('load');
  expect(wrapper.html()).toMatchSnapshot();
});

test('image load error', async () => {
  const wrapper = mount(Image, {
    props: {
      src: 'https://x',
      width: '100',
      height: '100',
      showError: true
    }
  });
  await wrapper.find('img').trigger('error');
  expect(wrapper.find('.nut-img-error').exists()).toBeTruthy();
  expect(wrapper.html()).toMatchSnapshot();
});

test('image loading', async () => {
  const wrapper = mount(Image, {
    props: {
      src: '',
      width: '100',
      height: '100',
      showLoading: true
    }
  });
  // await wrapper.find('img').trigger('load');
  expect(wrapper.find('.nut-img-loading').exists()).toBeTruthy();
  expect(wrapper.html()).toMatchSnapshot();
});

test('image render Round', async () => {
  const wrapper = mount(Image, {
    props: {
      src: '//img10.360buyimg.com/ling/jfs/t1/181258/24/10385/53029/60d04978Ef21f2d42/92baeb21f907cd24.jpg',
      width: '100',
      height: '100',
      round: true
    }
  });
  await wrapper.find('img').trigger('load');
  expect(wrapper.find('.nut-image-round').exists()).toBeTruthy();
  expect(wrapper.html()).toMatchSnapshot();
});
