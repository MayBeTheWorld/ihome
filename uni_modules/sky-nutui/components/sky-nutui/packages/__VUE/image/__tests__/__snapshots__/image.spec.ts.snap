// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image load error 1`] = `
"<view class=\\"nut-image\\" style=\\"width: 100px; height: 100px;\\"><img class=\\"nut-img\\" src=\\"https://x\\" alt=\\"\\" style=\\"object-fit: fill; object-position: center;\\">
  <!--v-if-->
  <view class=\\"nut-img-error\\"><i class=\\"nutui-iconfont nut-icon nut-icon-image-error\\" src=\\"\\"></i></view>
</view>"
`;

exports[`image loading 1`] = `
"<view class=\\"nut-image\\" style=\\"width: 100px; height: 100px;\\"><img class=\\"nut-img\\" src=\\"\\" alt=\\"\\" style=\\"object-fit: fill; object-position: center;\\">
  <view class=\\"nut-img-loading\\"><i class=\\"nutui-iconfont nut-icon nut-icon-image\\" src=\\"\\"></i></view>
  <!--v-if-->
</view>"
`;

exports[`image render 1`] = `
"<view class=\\"nut-image\\" style=\\"width: 100px; height: 100px;\\"><img class=\\"nut-img\\" src=\\"//img10.360buyimg.com/ling/jfs/t1/181258/24/10385/53029/60d04978Ef21f2d42/92baeb21f907cd24.jpg\\" alt=\\"\\" style=\\"object-fit: fill; object-position: center;\\">
  <!--v-if-->
  <!--v-if-->
</view>"
`;

exports[`image render Round 1`] = `
"<view class=\\"nut-image nut-image-round\\" style=\\"width: 100px; height: 100px;\\"><img class=\\"nut-img\\" src=\\"//img10.360buyimg.com/ling/jfs/t1/181258/24/10385/53029/60d04978Ef21f2d42/92baeb21f907cd24.jpg\\" alt=\\"\\" style=\\"object-fit: fill; object-position: center;\\">
  <!--v-if-->
  <!--v-if-->
</view>"
`;
